Monika After Story/game/screens.rpy

```diff
@@ -2628,12 +2640,83 @@ screen mas_gen_scrollable_menu(items, display_area, scroll_align, *args):
                     textbutton _(final_items[0]):
                         if final_items[2] and final_items[3]:
                             style "scrollable_menu_crazy_button"
+
                         elif final_items[2]:
                             style "scrollable_menu_new_button"
+
                         elif final_items[3]:
                             style "scrollable_menu_special_button"
+
+                        xsize display_area[2]
                         action Return(final_items[1])
 
+# Scrollable menu with checkboxes. Toggles values between True/False
+# Won't close itself until the user clicks on the return button
+
+# IN:
+#     items - list of tuples of the following format:
+#         (prompt, key, value, is_italic, is_bold)
```

Comment:
I would also change the way value works here. Given that if the value is a boolean and False, the item is disabled. I think return value and enabled should be differentiated using separate tuple items since there's a possibility we want things to start disabled but not necessarily return a bool.

Reply:
Agree, I left `is_italic` and `is_bold` only because that's how the gen menu works. Will change it for the new menu.

So basically you want the ability to set when the button will be checked. When its value is `True`, or when its value is `False`. Did I get it right?
> not necessarily return a bool

Sadly, I wasn't able to make it use anything, but `bool`s because It always ended up being quite spaghetti-ish. I thought If we would need to use different values to toggle, we could handle it upon getting return values from the screen. Do we really need it?

Reply:
Not quite what I mean. I'm thinking an ideal tuple being something like: `(prompt, key, value, enabled)`

The menu item being enabled when opening the menu should be separate from its return value.

Reply:
Alright, I'll change it!
Though, wouldn't it be confusing for people? You pass in `False` as value, but set it to be `selected`. The user sees that all the buttons are checked, and doesn't change anything. But as they clicks return, we will still get `False`.

Reply:
Perhaps `selected` isn't a good term then. Basically just something to indicate whether the option starts greyed out or not. Maybe change the `value` name to `return_value`

Reply:
Hmm, it's not so much `return_value`, as `default_value` or `current_value`. I'll need a new method to handle it anyway, lemme push it and we'll see if it's any better.

Reply:
Please, take a look and say if the new changes made it better.